src/Tablebot/Plugins/Join.hs

```diff
@@ -0,0 +1,68 @@
+-- |
+-- Module      : Tablebot.Plugins.Join
+-- Description : A command that adds a string between each word from an input.
+-- License     : MIT
+-- Maintainer  : tagarople@gmail.com
+-- Stability   : experimental
+-- Portability : POSIX
+--
+-- A command that outputs its input.
+module Tablebot.Plugins.Join (joinPlugin) where
+
+import Data.Text (Text, intercalate, pack, splitOn)
+import Discord.Types
+import Tablebot.Plugin
+import Tablebot.Plugin.Discord (Message, sendMessage)
+import Tablebot.Plugin.Parser (untilEnd)
+import Tablebot.Plugin.SmartCommand (Quoted (Qu), RestOfInput (ROI), parseComm)
+import Text.RawString.QQ
+
+joinify :: Text -> [Text] -> Text
+joinify sep text = sep <> " " <> (intercalate (" " <> sep <> " ") text) <> " " <> sep
+
+-- | @join@ outputs the second input interspersed with the first.
+join :: Command
+join = Command "join" (parseComm joinWithString)
+  where
+    joinWithString :: RestOfInput Text -> Message -> DatabaseDiscord ()
+    joinWithString (ROI t) m =
+      sendMessage m $
+        (\(sep : text) -> joinify sep text) $
+          splitOn " " t
+
+clap :: Command
+clap = Command "clap" (parseComm clappytime)
+  where
+    clappytime :: RestOfInput Text -> Message -> DatabaseDiscord ()
+    clappytime (ROI t) m =
+      sendMessage m $
+        joinify ":clap:" $
+          splitOn " " t
```

Comment:
You could do `clappytime :: Either (Quote Text, RestOfInput Text) RestOfInput Text -> Message -> DatabaseDiscord ()` to allow users to join with a string containing spaces:

`join "x y" a b c d e`
> x y a x y b x y c x y d x y e